strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-pypi-layout-provider/src/test/java/org/carlspring/strongbox/artifact/generator/PythonWheelArtifactGenerator.java

```diff
@@ -30,24 +34,59 @@
                                                    "\n" +
                                                    "Strongbox wheel package for test";
 
-    private String basedir;
+    private Path basedir;
+    
+    
+    public PythonWheelArtifactGenerator(Path basedir)
+    {
+        this.basedir = basedir;
+    }
 
     public PythonWheelArtifactGenerator(String basedir)
     {
-        this.basedir = basedir;
+        this.basedir = Paths.get(basedir);
+    }
+
+    @Override
+    public Path generateArtifact(String id,
+                                 String version,
+                                 int size)
+            throws IOException
+    {
+        PypiWheelArtifactCoordinates coordinates = new PypiWheelArtifactCoordinates(id,
+                                                                                    version,
+                                                                                    null,
+                                                                                    "py3",
+                                                                                    "none",
+                                                                                    "any");
+        return generateArtifact(coordinates);
     }
 
-    public void generateWheelPackage(PypiWheelArtifactCoordinates coordinates)
+    @Override
+    public Path generateArtifact(URI uri,
+                                 int size)
             throws IOException
     {
-        String packagePath = String.format("%s/%s-%s-py2-none-any.whl", basedir, coordinates.getId(),
+        PypiWheelArtifactCoordinates coordinates = PypiWheelArtifactCoordinates.parse(FilenameUtils.getName(uri.getPath()));
+        return generateArtifact(coordinates);
+    }
+
+    public Path generateArtifact(PypiWheelArtifactCoordinates coordinates)
+            throws IOException
+    {
+        String packagePath = String.format("%s-%s-py2-none-any.whl",
```

Comment:
@Firstmol can you please explain what means `none-any` in `packagePath`? 
Sorry, I am not familiar with Python, just want to be sure that we really don't have coordinates for it.

Reply:
python wheel packages have this naming pattern : `{distribution}-{version}(-{build tag})?-{python tag}-{abi tag}-{platform tag}.whl.`

I decided to use default value for `python-tag` , `abi-tag` and `paltform-tag`, because this  just test artifact(if you think it is necessary then I will do it). Also, I didn't find info about `build-tag`(no info how set this tag and how `buid-tag`  affects on METADATA file after build)

Reply:
fixed

Reply:
@Firstmol can you please do it like [this](https://github.com/strongbox/strongbox/pull/1291/files#r289666918)